### DISCUSSION TITLE
Looking for a 3rd Party API tutorial or template example.

### DISCUSSION THREAD

**Discussion by cngi1** [Budibase/budibase#5427] [Help]
I am struggling with "site in 15 min" advertisement.  

Is there a detailed implementation like the templates, none use a 3rd party API, that I can learn how the internals of budibase and external 3rd party API  data is used, referenced and the multitude of context changes in the budibase screen building? 

thanks in advance

**Comment by melohagan** [comment]
Hi @cngi1 

The REST API docs, which go through the basics, are here: https://docs.budibase.com/docs/rest

For the screen building you can browse through the design section of the docs: https://docs.budibase.com/docs/screens

If you're having a look through the templates, you should be able to reasonably interchange the datasources with your api queries.

Tutorials are something we do want to improve upon, but if there's any specific challenges you're facing, these discussions are a good resource.

**Comment by cngi1** [comment]
i have reviewed that info and find it lacking.  

looking at how the templates work gives hints but to expect a new person to make a transition to API is just plain lazy on the past of documentation staff.

obviously it must be harder since you provide zero examples in templates or any full detailed example in the Api docs.

so in short no it is not enough. please provide working 3 party API example of any template you offer. that certainly can not be too much to ask for one complete example????

**Reply by shogunpurple** [reply]
@cngi1 - I would implore you to revise your tone. I’ll remind you that the budibase community forum is completely free of charge and we respond to requests as quickly as we can. @melohagan has been nothing but helpful to you, responding to your many requests.

Paid support contracts are available for customers who need dedicated support, and we are more than happy to discuss this. 

I’m not sure what the goal is here - but I can guarantee it will not get you the result you are looking for. We are working hard on improving our documentation and tutorials as we mentioned, and as you could imagine there are many more things we are working on also.

We are happy to help you with your use case if you tell us more about what exactly you want to do - do you want to display data from an API in a table, or hit API endpoints for create, read, update and delete operations?

Thank you in advance.

**Reply by cngi1** [reply]
My business operates with users on a very complicated testing platform. Every feature we offer requires a thorough explanation of how to use it and demonstration examples to enable our customers to use each feature of our platform. We don't offer it otherwise because it frustrates the users ,they may just give up and cost us in the long run. 

I am sorry that you feel pointing out a shortcoming in the BB knowledge base has hit a nerve. But truth is still truth. If this was my company I would have answered these comments with a thank you. for pointing this out.  I would have taken the 15 minutes that an expert needs to create  a short demo / template for guidance.   I would have made time since no other example is present and certainly a hole in the documentation. 

so, 
A novice customer said  I am new to your product and I don't understand how to convert a template, using local data, to one using an API . The only answer provided here was look at the same info and figure it out yourself.  

So I hope the tone does change the one directed at new users.

Your welcome.

**Reply by shogunpurple** [reply]
That's great - congratulations. 

Pointing out an issue in the budibase knowledge base is not what has hit a nerve - as mentioned, it's something that we are working hard on improving, it's on our radar and the feedback is taken into account. It was your tone - which is objectively rude, to someone who is trying to help you. We are always appreciative of feedback and address as much as we possibly can.

Regardless - I'll ask again:
> We are happy to help you with your use case if you tell us more about what exactly you want to do - do you want to display data from an API in a table, or hit API endpoints for create, read, update and delete operations?

If you tell us what exactly your use case is, we can put together and example to show you how exactly to build it, and provide you with the support you need. What API do you need to integrate with? What do you want to do with the API? Forms? Charts? Any information around this would be really helpful.

Thanks

**Reply by cngi1** [reply]
Your opinion noted but not agreed with so moving on.

I rejected the request for use case information because the project API is not user-friendly and far too extensive for me to expect anyone at BB to devote time to it. It'll take a lot longer than doing a demo.

Transformers will be required for everything. Transformers that convert XML to JSON and back to XML. The JSON that is returned is almost never flat, and it always has a lot of nested levels. Some calls only accept and return XML. 

1. Read a dashboard with hundreds of status updates and API links into BB page(s) for display, analysis, charting, and redirection to the appropriate admin page for each item represented by the status.

2. admin each of these items (edit, delete, add, and update) and update, add, and delete using the API. (XML->JSON->XML) Files will need to be transmitted as part of multipart/form-data in this critical path use case.
  note here: https://github.com/Budibase/budibase/discussions/5511

3. Everything will require transformers, and the big new user challenge will be moving that information thru all the steps and forms for editing, deleting, and adding items.

But if you're still interested, I can email you a link to the Swagger API. You'll opt to do a demo instead after seeing it.

Is there a way to send private info?  I would start with sending the swagger link just to look at.

thank you

**Comment by cngi1** [comment]
any hope here for a tutorial?
